src/strategies/offer_forwarder/PooledForwarder.sol

```diff
@@ -0,0 +1,227 @@
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+pragma solidity ^0.8.10;
+
+pragma abicoder v2;
+
+import {Forwarder} from "src/strategies/offer_forwarder/abstract/Forwarder.sol";
+import {IMakerLogic} from "src/strategies/interfaces/IMakerLogic.sol";
+import {AaveRouter} from "src/strategies/routers/AaveRouter.sol";
+import {IERC20, MgvLib} from "src/MgvLib.sol";
+import {IMangrove} from "src/IMangrove.sol";
+import {AbstractRouter} from "src/strategies/routers/AbstractRouter.sol";
+import {TransferLib} from "src/strategies/utils/TransferLib.sol";
+
+//Using a AaveDeepRouter, it will borrow and deposit on behalf of the reserve - but as a pool, and keep a contract-local balance to avoid spending gas
+// gas overhead of the router for each order:
+// - supply ~ 250K
+// - borrow ~ 360K
+//This means that yield and interest should be handled for the reserve here, somehow.
+contract PooledForwarder is IMakerLogic, Forwarder {
+  // balance of token for an owner - the pool has a balance in aave and on `this`.
+  mapping(IERC20 => mapping(address => uint)) internal ownerBalance;
+
+  constructor(IMangrove mgv, address deployer, address aaveAddressProvider)
+    Forwarder(mgv, new AaveRouter(aaveAddressProvider, 0, 1 , 700_000), 30_000)
+  {
+    AbstractRouter router_ = router();
+    router_.bind(address(this));
+    setAdmin(deployer);
+    router_.setAdmin(deployer); // consider if admin should be this contract?
+  }
+
+  // Increases balance of token for owner.
+  function increaseBalance(IERC20 token, address owner, uint amount) private returns (uint) {
+    uint newBalance = ownerBalance[token][owner] + amount;
+    ownerBalance[token][owner] = newBalance;
+    return newBalance;
+  }
+
+  // Decrease balance of token for owner.
+  function decreaseBalance(IERC20 token, address owner, uint amount) private returns (uint) {
+    uint currentBalance = ownerBalance[token][owner];
+    require(currentBalance >= amount, "PooledForwarder/decreaseBalance/amountMoreThanBalance");
```

Comment:
I am afraid the revert reason is too long and will be truncated by Mangrove when cast to 32 bytes